sonar-scanner-engine/src/main/java/org/sonar/scanner/issue/ignore/scanner/IssueExclusionsRegexpScanner.java

```diff
@@ -19,93 +19,60 @@
-import org.sonar.api.batch.ScannerSide;
-import org.sonar.scanner.issue.ignore.pattern.IssueExclusionPatternInitializer;
-import org.sonar.scanner.issue.ignore.pattern.IssuePattern;
-import org.sonar.scanner.issue.ignore.pattern.LineRange;
-
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.nio.charset.Charset;
-import java.nio.file.Files;
-import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.regex.Pattern;
 
-@ScannerSide
-public class IssueExclusionsRegexpScanner {
-
-  private static final Logger LOG = LoggerFactory.getLogger(IssueExclusionsRegexpScanner.class);
-
-  private IssueExclusionPatternInitializer exclusionPatternInitializer;
-  private List<java.util.regex.Pattern> allFilePatterns;
-  private List<DoubleRegexpMatcher> blockMatchers;
-
-  // fields to be reset at every new scan
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.batch.fs.internal.FileMetadata.CharHandler;
+import org.sonar.scanner.issue.ignore.pattern.LineRange;
+import org.sonar.scanner.issue.ignore.pattern.PatternMatcher;
+import org.sonar.scanner.issue.ignore.scanner.IssueExclusionsLoader.DoubleRegexpMatcher;
+
+public class IssueExclusionsRegexpScanner extends CharHandler {
+  private static final Logger LOG = LoggerFactory.getLogger(IssueExclusionsLoader.class);
+
+  private final StringBuilder sb = new StringBuilder();
+  private final List<Pattern> allFilePatterns;
+  private final List<DoubleRegexpMatcher> blockMatchers;
+  private final String componentKey;
+  private final PatternMatcher patternMatcher;
+
+  private int lineIndex = 1;
+  private List<LineExclusion> lineExclusions = new ArrayList<>();
+  private LineExclusion currentLineExclusion = null;
+  private int fileLength = 0;
   private DoubleRegexpMatcher currentMatcher;
-  private int fileLength;
-  private List<LineExclusion> lineExclusions;
-  private LineExclusion currentLineExclusion;
 
-  public IssueExclusionsRegexpScanner(IssueExclusionPatternInitializer patternsInitializer) {
-    this.exclusionPatternInitializer = patternsInitializer;
-
-    lineExclusions = new ArrayList<>();
-    allFilePatterns = new ArrayList<>();
-    blockMatchers = new ArrayList<>();
-
-    for (IssuePattern pattern : patternsInitializer.getAllFilePatterns()) {
-      allFilePatterns.add(java.util.regex.Pattern.compile(pattern.getAllFileRegexp()));
-    }
-    for (IssuePattern pattern : patternsInitializer.getBlockPatterns()) {
-      blockMatchers.add(new DoubleRegexpMatcher(
-        java.util.regex.Pattern.compile(pattern.getBeginBlockRegexp()),
-        java.util.regex.Pattern.compile(pattern.getEndBlockRegexp())));
-    }
-
-    init();
+  IssueExclusionsRegexpScanner(String componentKey, List<java.util.regex.Pattern> allFilePatterns, List<DoubleRegexpMatcher> blockMatchers, PatternMatcher patternMatcher) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this fully qualified name with "Pattern" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1942)